cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/StreamUtils.java

```diff
@@ -425,12 +427,24 @@ public Long getResult() {
   }
 
   /**
-   * Gets a table name for stream consumer state stores within a given namespace.
+   * Gets a TableId for stream consumer state stores within a given namespace.
+   * @param namespace the namespace for which the table is for.
+   * @return constructed TableId
+   */
+  public static TableId getStateStoreTableId(Id.Namespace namespace) {
+    String tableName = String.format("%s.%s.state.store",
+                                     Constants.SYSTEM_NAMESPACE, QueueConstants.QueueType.STREAM.toString());
+    return TableId.from(namespace.getId(), tableName);
+  }
+
+  /**
+   * Gets the table name for stream consumer state stores within a given namespace.
    * @param namespace the namespace for which the table is for.
    * @return constructed table name
    */
   public static String getStateStoreTableName(Id.Namespace namespace) {
```

Comment:
Each of them should do their own logic to translate TableId into something it needs (e.g. for leveldb, it can use namespace as directory and name as sub-directory, while for in-memory, it just concat ns-name and use it as a String to lookup a map).